### ISSUE TITLE
UnicodeDecodeError on Response body

### ISSUE THREAD

**Issue by benjimin** [pyppeteer/pyppeteer#84]
Unable to obtain Response body for requests of non-text objects, such as images, as `Response.json()` and `Response.text()` throw UnicodeDecodeErrors. The following snippet produces output including `gif` and `png`:
```python
browser = await pyppeteer.launch()
try:
    page = await browser.newPage()

    @page.on('requestfinished')
    async def handler(r):
        if r.response.status == 200:
            try:
                data = await r.response.text()
            except UnicodeDecodeError:
                print(r.url.split('.')[-1])
    
    await page.goto('https://www.google.com.au')
    await page.waitFor(3000)
finally:
    await browser.close()
```
It seems like something is attempting to interpret binary data as utf8 (and then complaining that it isn't valid utf8). The tracebacks include:
```
File "[..]site-packages/pyppeteer/network_manager.py", line 673, in text
    return content.decode('utf-8')
```
Tested on pyppeteer 0.0.25, MacOS.

**Comment by Mattwmaster58** [comment]
I tested this on the `pup2.1.1` branch and got the following result:

```
data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABYAAAAWCAYAAADEtGw7AAAAoklEQVR42mMwTvvPQAZ2AOKZQPwAimdCxeBqyDX0DBD/R8NnkA0nx+CZWAyF4ZmUGPwAj8EPBp3BzkB8mdSgwBfTMkC8EsmAW8RGHq6YPgu14DOUD6IrgNiF2OSGL6ZheDUQyxEbbMREyCcgdiU1kkmKaXIMxhcUlylxMa7Iu0VpGONKbqDYL6ckVRDCZKdjmua84VkI0bQ8plkNQrM6jygMAODsqWK79M8/AAAAAElFTkSuQmCC
gif
webp
png
png
webp
```

I'm linking the PR to merge that branch as it seems to completely fix this problem.

**Comment by benjimin** [comment]
Exactly, note the output only prints because an exception is thrown. 

I've noticed that `Response.buffer()` does seem to work, so perhaps I should have asked first, what is the expected result if invoking something like `Response.json()['body']` for a request of an image? Is that intended to fail or succeed?

**Comment by Mattwmaster58** [comment]
Sorry, I must admit didn't read into your reproduction code all that carefully

**Comment by Mattwmaster58** [comment]
> I've noticed that Response.buffer() does seem to work, so perhaps I should have asked first, what is the expected result if invoking something like `Response.json()['body']` for a request of an image? Is that intended to fail or succeed?

Presumably that would fail, here's the JS implementation:
```js
  async json() {
    const content = await this.text();
    return JSON.parse(content);
  }
```

**Comment by Mattwmaster58** [comment]
The issue stems from the fact that we don't tolerate errors in UTF-8 decoding, and apparently node's `Buffer.toString` does, but of course that isn't mentioned anywhere in the docs. I've pushed up a fix for this on the pup2.1.1 branch.
